security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -170,8 +194,13 @@ func (sf *SecretFetcher) scrtAdded(obj interface{}) {
 		return
 	}
 
-	t := time.Now()
 	resourceName := scrt.GetName()
+	if !isIngressGatewaySecret(scrt) {
+		log.Debugf("secret %s is not an ingress gateway secret, skip adding secret", resourceName)
+		return
+	}
+
+	t := time.Now()
 	newCert, newKey, newRoot, valid := extractCertAndKey(scrt)
```

Comment:
probably move the logic in ```isIngressGatewaySecret``` into ```extractCertAndKey``` (by adding another return param indicate whether suppress logging) ? so that the logic would be in on place.

Reply:
Thanks for the comment. extractCertAndKey is intended to check ingress gateway secret, and a warning message would be created if extractCertAndKey returns false. We want to avoid those warning messages for non-ingress gateway secret, but still keep it for invalid ingress secret.